src/lib/universal-client.js

```diff
@@ -0,0 +1,638 @@
+                    + json.v + ') used!');
+
+              // send first request
+              thisCometVisu.running = true;
+              if (thisCometVisu.initialAddresses.length) {
+                this.xhr = $.ajax({
+                  url : thisCometVisu.getResourcePath("read"),
+                  dataType : 'json',
+                  context : this,
+                  data : thisCometVisu
+                      .buildRequest(thisCometVisu.initialAddresses)
+                      + '&t=0',
+                  success : this.handleReadStart,
+                  beforeSend : this.beforeSend
+                });
+              } else {
+                // old behaviour -> start full query
+                this.xhr = $.ajax({
+                  url : thisCometVisu.getResourcePath("read"),
+                  dataType : 'json',
+                  context : this,
+                  data : thisCometVisu.buildRequest() + '&t=0',
+                  success : this.handleRead,
+                  error : this.handleError,
+                  beforeSend : this.beforeSend
+                });
+              }
+            },
+            /**
+             * This function gets called once the communication is established
+             * and session information is available
+             * 
+             * @method handleRead
+             * @param json
+             */
+            handleRead : function(json) {
+              if (!json && (-1 == this.lastIndex)) {
+                if (thisCometVisu.running) { // retry initial request
+                  this.retryCounter++;
+                  this.xhr = $.ajax({
+                    url : thisCometVisu.getResourcePath("read"),
+                    dataType : 'json',
+                    context : this,
+                    data : thisCometVisu.buildRequest() + '&t=0',
+                    success : this.handleRead,
+                    error : this.handleError,
+                    beforeSend : this.beforeSend
+                  });
+                  this.watchdog.ping();
+                }
+                return;
+              }
+
+              if (json && !this.doRestart) {
+                this.lastIndex = json.i;
+                var data = json.d;
+                this.readResendHeaderValues();
+                thisCometVisu.update(data);
+                this.retryCounter = 0;
+              }
+
+              if (thisCometVisu.running) { // keep the requests going
+                this.retryCounter++;
+                this.xhr = $.ajax({
+                  url : thisCometVisu.getResourcePath("read"),
+                  dataType : 'json',
+                  context : this,
+                  data : thisCometVisu.buildRequest() + '&i=' + this.lastIndex,
+                  success : this.handleRead,
+                  error : this.handleError,
+                  beforeSend : this.beforeSend
+                });
+                this.watchdog.ping();
```

Comment:
Running the code (eibd backend, no change to my config, just a local pull of this request) I get a error in this line:
`Uncaught TypeError: this.watchdog.ping is not a function`

(Why? I don't know yet...)